comp/dogstatsd/server/server.go

```diff
@@ -605,6 +609,12 @@ func (s *server) handleMessages() {
 }
 
 func (s *server) restoreBlocklistFromLocalConfig() {
+	s.log.Debug("Restoring blocklist with local config.")
+
+	// update the runtime config to be consistent
+	// in `agent config` calls.
+	s.config.Set("statsd_metric_blocklist", s.localBlocklistConfig.metricNames, model.SourceAgentRuntime)
```

Comment:
I think the correct source to set here is what the source of the blocklist was before the blocklist was set via RC (if it was set via RC).

I don't think the behavior inline will lead to bugs today, but I feel it doesn't fit well with how the config package is designed 🤔